civis/futures.py

```diff
@@ -279,11 +311,18 @@ def cancel(self):
             elif not self.done():
                 # Cancel the job and store the result of the cancellation in
                 # the "finished result" attribute, `_result`.
-                self._result = self.client.scripts.post_cancel(self.job_id)
+                try:
+                    self._result = self.client.scripts.post_cancel(
+                        self.job_id)
+                except CivisAPIError:
```

Comment:
If the job has never been run, you cannot post a cancel IIRC.

Reply:
That's true. I think it would be clearer to use an `if self.run_id is not None`, though. I'm worried about masking other API errors which would leave a run not cancelled.

Reply:
Here is the response

```python
$ civis scripts post-cancel <ID>
code: 404
error: not_found
errorDescription: The requested resource could not be found.
```